include/boost/fusion/container/vector/vector_fwd.hpp

```diff
@@ -9,11 +9,39 @@
 
 #include <boost/config.hpp>
 #include <boost/fusion/support/config.hpp>
+#include <boost/fusion/container/vector/detail/config.hpp>
 
 ///////////////////////////////////////////////////////////////////////////////
 // With no variadics, we will use the C++03 version
 ///////////////////////////////////////////////////////////////////////////////
+#if !defined(BOOST_FUSION_HAS_VARIADIC_VECTOR)
 # include <boost/fusion/container/vector/detail/cpp03/vector_fwd.hpp>
+#else
 
+///////////////////////////////////////////////////////////////////////////////
+// C++11 interface
+///////////////////////////////////////////////////////////////////////////////
+#include <boost/preprocessor/cat.hpp>
+#include <boost/preprocessor/repetition/repeat.hpp>
+#include <cstddef>
+
+namespace boost { namespace fusion
+{
+    template <typename ...T>
+    struct vector;
+
+    template <std::size_t N>
+    struct numbered_vector_tag;
+
+#define FUSION_VECTOR_N_ALIASES(z, N, d)                                    \
+    template <typename ...T>                                                \
+    using BOOST_PP_CAT(vector, N) = vector<numbered_vector_tag<N>, T...>;
+
+    BOOST_PP_REPEAT(51, FUSION_VECTOR_N_ALIASES, ~)
```

Comment:
Keep in mind that vectorN is needed only because of the lack of variadics in C++03. So I had to somehow map each concrete instantiation of vectorN to a vector. There can be a better implementation here. vectorN is a C++03 implementation artefact, IMO.